src/Renderer/LayeredMaterial.js

```
@@ -244,83 +283,87 @@ LayeredMaterial.prototype.setTexture = function(texture, layer, slot, pitScale)
-        }
-    }
+LayeredMaterial.prototype.setLightingOn = function(enable) {
+    this.uniforms.lightingOn.value = enable;
 };
 
-LayeredMaterial.prototype.setLightingOn = function(enable) {
-    this.uniforms.lightingOn.value = enable === true ? 1 : 0;
+LayeredMaterial.prototype.setLayerFx = function(id, fx) {
+        this.uniforms.paramLayers.value[id].z = fx;
 };
 
-LayeredMaterial.prototype.setLayerOpacity = function(id, opacity) {
+LayeredMaterial.prototype.setLayerOffset = function(id, offset) {
+        this.uniforms.paramLayers.value[id].x = offset;
+};
 
-    if (this.paramLayers[id]) {
-        this.paramLayers[id].w = opacity;
-    }
+LayeredMaterial.prototype.setLayerUV = function(id, idUV) {
+        this.uniforms.paramLayers.value[id].y = idUV;
+};
+
+LayeredMaterial.prototype.setLayerOpacity = function(id, opacity) {
+    if(this.uniforms.paramLayers.value[id])
+        this.uniforms.paramLayers.value[id].w = opacity;
 };
 
 LayeredMaterial.prototype.setLayerVisibility = function(id, visible) {
+        this.uniforms.visibility.value[id] = visible;
+};
 
-    if (this.paramLayers[id]) {
-        this.paramLayers[id].z = visible ? 1 : 0;
-    }
+LayeredMaterial.prototype.setLayerNbTextures = function(id, nb) {
+        this.nbTexturesLayers[id] = nb;
+};
+
+LayeredMaterial.prototype.nbLoadedTextures = function() {
+    return this.nbTextures[ELEVATION] + this.nbTextures[COLOR];
 };
 
-LayeredMaterial.prototype.setNbLayersColor = function(n) {
-    this.uniforms.nColorLayer.value = n;
+LayeredMaterial.prototype.getNumberColorLayer = function() {
+    return this.layers.length;
 };
 
-LayeredMaterial.prototype.getLevelLayerColor = function(id, slot) {
+LayeredMaterial.prototype.getLevelLayerColor = function(id, layer) {
 
-    var level = this.Textures[id][slot || 0].level;
+    var index = this.layers.indexOf(layer);
+    var slot = index > -1 ? this.getOffsetByIndex(index) : 0;
```

Maybe we should stop silently hiding errors.
`inded < 0` is a programming error so we should notify the programmer (`throw` maybe?)